### PR TITLE
Add P-space property

### PR DESCRIPTION
**Pull request by hpecora1**
Addresses issue #220

### CODE REVIEW

**Comment by StevenClontz** [comment]
Hayden and I worked together on this just now - @prabau let us know if you have any suggestions for improvement.

**Comment by prabau** [comment]
Cool.  I'll take a look.

File: properties/P000147.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+uid: P000147
+name: P-space
+aliases: 
+  - pseudo-discrete
+refs:
+  - doi: 10.1016/0016-660X(72)90026-8
+    name: A topological view of P-spaces (A. Misra)
+---
+
+The intersection of countably many open sets is open. Defined in {{doi:10.1016/0016-660X(72)90026-8}} with the additional assumption of {P2}.
```
Comment (prabau) [inline properties/P000147.md:11]: Just to be completely clear, how about "... with the additional assumption of {P2}, which is not assumed here".

Comment (StevenClontz) [reply properties/P000147.md:11]: @hpecora1 if you agree, you can add a commit to this effect. There's two ways to do it: use the `github.dev` interface we used earlier today, or you can directly edit this file by going to "view reviewed changes" and using the menu to select "Edit this file".

**Comment by prabau** [comment]
I don't think the alias "pseudo-discrete" is justified.  In his introduction Misra mentions that P-space was an abbreviation for "pseudo-discrete space" in Gillman and Henriksen, but as Misra mentions, their notion of P-space was slightly different than the notion taken as definition by Misra.

**Comment by prabau** [comment]
For the definition of P147, I think it would be good to add equivalent formulations that seem useful for verifying the property.  Something like the following for example:

> Equivalently, a space in which every countable union of closed sets is closed.
>
> Equivalently, a space in which every point is a P-point, where a point $x$ is called a P-point if every countable intersection of neighborhoods of $x$ is a neighborhood of $x$.

**Comment by prabau** [comment]
For T351, instead of Corollary 3.3, it should refer to the proof of Proposition 3.2.  But that proof does not use the T2 or T1 property anywhere, so the more generally valid result is [regular + P-space ==> zero-dim]

File: spaces/S000178/properties/P000004.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+space: S000178
+property: P000004
+value: true
+refs:
+  - doi: 10.1016/0016-660X(72)90026-8
+    name: A topological view of P-spaces (A. Misra)
+---
+
+Shown in Example 3.1 of {{doi:10.1016/0016-660X(72)90026-8}}.
```
Comment (prabau) [inline spaces/S000178/properties/P000004.md:10]: "Shown in ...": nothing is shown in the paper, it is just claimed there (not that it's necessarily difficult).  So how about just saying "See Example ..."?

(same comment applies to the other properties)

**Comment by StevenClontz** [comment]
> For T351, instead of Corollary 3.3, it should refer to the proof of Proposition 3.2. But that proof does not use the T2 or T1 property anywhere, so the more generally valid result is [regular + P-space ==> zero-dim]

I suspected that might be an easy improvement, but did not take the time during our call earlier to look closely. Now that I have, I agree that T1 is clearly unnecessary and citing the rpoof of Proposition 3.2 is a better reference.

**Comment by prabau** [comment]
The name "Misra subspace" for S178 does not seem very good.  Can we come up with something better?
Also, the README should have a link to S177 that it is a subspace of.

"Misra space" for S177 is not too descriptive either.  "Misra's $E_0$ P-space" or something else?  Not sure.  (The article also has example $E_\omega$ in 5.1 if one would ever care to add it in the future, mentioned in https://math.stackexchange.com/questions/3765829).

@StevenClontz What's your take on the following? I'll tell you my opinion, but it's up for discussion.
When providing the definition of an example space, I think it's worth spelling it out in full if the description is relatively simple, or if it's necessary to refer to specific element when justifying the properties.  On the other hand if the construction is very involved, just a reference would be enough.  It's a judgement call.  But at least we can also provide more context about the main purpose of the example.

So in the README for S177 we could mention that Misra's 3.1 provides an example of P-space that is T2, semiregular, but not T2.5.  And similarly for S178.

File: spaces/S000178/properties/P000147.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+---
+space: S000178
+property: P000147
+value: true
+refs:
+  - doi: 10.1016/0016-660X(72)90026-8
+    name: A topological view of P-spaces (A. Misra)
+---
+
+Shown in Example 3.1 of {{doi:10.1016/0016-660X(72)90026-8}}.
```
Comment (prabau) [inline spaces/S000178/properties/P000147.md:10]: Possible justification: It's a subspace of S177 and the P-space property is preserved by subspaces.

**Comment by prabau** [comment]
Not sure if you want to defer this to a later PR, but S17 (cocountable topology on the reals) is a simple example of P-space.

**Comment by StevenClontz** [comment]
@prabau lest this get too stale, would you like to make your suggested improvements? Then I'll do the final review and (presumably) approve and merge.

**Comment by prabau** [comment]
@hpecora1 I have added changes based on the review.  Please see if that makes sense.

**Comment by StevenClontz** [comment]
I'll approve if @hpecora1 is happy with the changes.

**Review by StevenClontz — APPROVED** [review]

**Pull request merged**